src/embedding/kd.ts

```diff
@@ -332,6 +358,53 @@ export namespace EmbeddedKpiDashboard {
         return getEventType(obj) === GdcKdCommandType.SetSize;
     }
 
+    /**
+     * Data type of SetFilterContext command
+     */
+    export type SetFilterContextCommandData = IGdcKdMessageEnvelope<GdcKdCommandType.SetFilterContext, EmbeddedGdc.IFilterContextContent>;
+
+    /**
+     * Add or update the filter context
+     *
+     * Contract:
+     * - if filters is same with filters on the KD filter bar, then update the filters on the filter bar and apply the filters to dashboard
+     * - if filters is new one, then add it to the KD filter bar and apply to dashboard
```

Comment:
1. `filters are...`
2. `... add them ...`. Note that this is for KD edit mode only

Reply:
```
if filters existed on the KD filter bar, then the filters will be updated and the insight on dashboard will be applied
if filters are new one, then the the filters will be added and the insight on dashboard will be applied
```